frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Outlet } from 'react-router-dom';
+
+import { AppRoute } from '~/libs/enums/app-route.enum';
+import { useEffect, useNavigate } from '~/libs/hooks/hooks.js';
+import { storage, StorageKey } from '~/libs/packages/storage/storage.js';
+
+const ProtectedRoute = (): React.ReactElement => {
+  const navigate = useNavigate();
+  useEffect(() => {
+    storage
+      .get(StorageKey.TOKEN)
+      .then((token) => {
+        if (!token) {
+          navigate(AppRoute.SIGN_IN);
+        }
+      })
+      .catch((error) => {
+        const typedError = error as Error;
+        throw new Error(typedError.message);
+      });
+  }, [navigate]);
```

Comment:
```suggestion
```

too complicated. you can simplify it

```ts
const { user } = useSelector((user) => user);
const hasUser = Boolean(user);

if (!hasUser) {
  // redirect
}

// component
```
(this is only an example. Adapt for our application)

Reply:
+ with current implementation anything manually set into localStorage `token` key may lead to uncontrolled access to ProtectedRoute